### PR TITLE
JavaScript capstone project - Your API-based webapp

### PR DESCRIPTION
**Pull request by MohammadYaser**
1. Use JavaScript to make websites dynamic and build basic single page apps.
2. Use ES6 syntax.
3. Use ES6 modules.
4. Use callbacks and promises.
5. Use webpack.
6. Apply JavaScript best practices and language style guides in code.
7. Use AAA pattern for unit tests.
8. Write units tests for a JavaScript app.
9. Follow Gitflow.
10. Solve simple git conflicts.
11. Send and receive data from an API.
12. Use API documentation.
13. Understand and use JSON.
14. Make JavaScript code asynchronous.
15. Perform a code review for a team member.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+const setupHeaderLinkListeners = () => {
+  menuLinks.forEach((link) => {
+    const { dataset: { category } } = link;
+    link.addEventListener('click', async (e) => {
+      e.preventDefault();
+      await renderMealCategory(category);
+    });
+  });
```
Comment (mikethreels) [inline src/index.js:23]: - [ ] when trying to add a comment to a meal the user receives the below error, please make sure to fix this before requesting another review.

image

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job getting the info from the API.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (MohammadYaser) [reply src/index.js:23]: Hi @mikethreels 
we both have checked the problem you have mentioned but we didn't find any issue like this,
even we have added some new commends to several meals:
image

File: src/modules/popupModal.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  newBtnAddComment.addEventListener('click', async () => {
+    const username = newUsernameInput.value;
+    const comment = newCommentInput.value;
+
+    if (username !== '' && comment !== '') {
+      await submitComment(id, username, comment);
+      newUsernameInput.value = '';
+      newCommentInput.value = '';
+      closeModal();
+    }
```
Comment (elisha2kyakpo1) [inline src/modules/popupModal.js]: - [ ] I tested your app by commenting on different cards. When I type text in the input fields and press the add comment button, the form does nothing, but when I refresh the page I can see my comment. This issue is not a good user experience. Kindly troubleshoot and fix this issue, You can delete the app on your machines and clone it from GitHub.

comment

File: src/index.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <nav>
+            <div class="logo-container">
+                <img src="./logo.png" alt="Logo">
+            </div>
+            <ul class="menu">
+                <li><a href="#" data-category="Seafood">Seafood</a><span></span></li>
+                <li><a href="#" data-category="Lamb">Lamb</a><span></span></li>
+                <li><a href="#" data-category="Chicken">Chicken</a><span></span></li>
+                <li><a href="#" data-category="Vegetarian">Vegetarian</a><span></span></li>
+            </ul>
```
Comment (elisha2kyakpo1) [inline src/index.html:21]: - [ ] [OPTIONAL] You may have forgotten to add the items counter on the navigation menu.

counters

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @MohammadYaser, @Yonasle, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

👍 You did well to communicate on the PR

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (algerina) [inline src/index.js:23]: - [ ] You still have an issue with the comments, when trying to add a comment the app is crushing and the comments are repeated on all the cards. Kindly fix the issue.

File: README.md [hunk 4]
```diff
@@ -1 +1,140 @@
...
+<br>
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+
+## 🚀 Live Demo & Video presentation <a name="live-demo"></a> <a name='video'></a>
+
+Link : https://drive.google.com/file/d/1n02orUvy-Z2FwmTrGpj6lo2k8-tb2u4G/view?usp=drive_link
+<br>
```
Comment (algerina) [inline README.md:52]: - [ Optional] There is no link to the live version.

**Review by algerina — CHANGES_REQUESTED** [review]
Hi team ,

Great job so far! 

May I highlight the following: 
-  Your app is almost working well ✔
- Well done adding the video presentation✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

File: src/modules/popupModal.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            <div class="commentBox-beautiful">
+              <p class="comment-name">${comment.username}:</p>
+              <p class="commentText-beautiful">${comment.comment}</p>
+              <p class="date-beautiful">${comment.creation_date}</p>
+            </div>
+          `,
+        )
+        .join('');
+    }
+  };
```
Comment (abdulhamiid) [inline src/modules/popupModal.js]: - [ ] As requested by the previous reviewers, please there are still some issues we need to fix 😊

- [ ] When the Comments popup loads, the web app is expected to retrieve data from the selected API and show details about the selected item. 

- [ ] And likewise, retrieve data from the [Involvement API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270) to show the item comments.

- [ ] In your app, the comments are being fetched continuously and comments from other cards are also displayed.

- [ ] Kindly check the loom [video](https://www.loom.com/share/c8f985e008424b4d95bb82deb8f9f4e1?sid=8238f019-83d2-45f2-ac69-56189434eb4e) I have recorded on this.

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
Hi Team 👋, 

almost-there

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED ✔️ 
Hi @MohammadYaser , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Alt Text
Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**